### PR TITLE
Add string escaping JSON tests to the test_json_matrix

### PR DESCRIPTION
**Pull request by revans2**
This contributes to #10491 and I intend to add more here, but this is just to put up what I have done so far as kind of a checkpoint.

### CODE REVIEW

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
I think this is good enough for now. I will go back through #10491 and add more tests on other PRs as needed.

File: integration_tests/src/main/python/json_matrix_test.py [hunk 1]
```diff
@@ -895,7 +906,8 @@ def test_from_json_doubles(std_input_path, input_file):
     pytest.param("boolean_formatted.json", marks=pytest.mark.xfail(reason='https://github.com/rapidsai/cudf/issues/15260')),
     pytest.param("int_array_formatted.json", marks=pytest.mark.xfail(reason='https://github.com/NVIDIA/spark-rapids/issues/10573')),
     "int_struct_formatted.json",
-    pytest.param("int_mixed_array_struct_formatted.json", marks=pytest.mark.xfail(reason='https://github.com/rapidsai/cudf/issues/15260'))])
+    pytest.param("int_mixed_array_struct_formatted.json", marks=pytest.mark.xfail(reason='https://github.com/rapidsai/cudf/issues/15260')),
+    pytest.param("escaped_strings.json", marks=pytest.mark.xfail(reason='https://github.com/rapidsai/cudf/issues/15260'))])
```
Comment (abellina) [inline integration_tests/src/main/python/json_matrix_test.py:910]: just to double check, is this the right github issue for the escaped_strings xfail? Should it not be: https://github.com/NVIDIA/spark-rapids/issues/10196

Comment (abellina) [reply integration_tests/src/main/python/json_matrix_test.py:910]: same comment for the ones below

Comment (revans2) [reply integration_tests/src/main/python/json_matrix_test.py:910]: Yes it is. It does not look that way on the surface, but it is failing with an error

```
column_factories.hpp:343: Invalid, non-fixed-width type.
```

mixed_type_as_string means that it sees multiple different types and tries to return the value as a string. Here it looks like the values are already strings and it is the same error. If you want me to file a separate issue I can.

Comment (abellina) [reply integration_tests/src/main/python/json_matrix_test.py:910]: No that's fine. Thanks

**Review by abellina — APPROVED** [review]

**Pull request merged**